### ISSUE TITLE
Disable scroll but allow tap on item

### ISSUE THREAD

**Issue by federicomiralles** [kizitonwose/CalendarView#38]
@kizitonwose I'm stuck with the following situacion...
On one hand I need to disable the horizontal scrolling on the calendar as shown on here: https://github.com/kizitonwose/CalendarView/issues/13 and do it programatically.
But at the same time I need to be able to tap/click on days of the calendar (by setting a click listener on the DayViewContainer view):
`view.setOnClickListener(new View.OnClickListener()...`
The problem is that by disabling the scroll I'm intercepting the Action.MOVE used by the taps as well, therefore onInterceptTouchEvent is swallowing the event, hence not allowing it to reach the defined view click listener.

So to sum up I need to disable the swiping of the calendar but allowing taps of each of days of the calendar...

Any suggestion?

Thanks in advance,

Federico

**Comment by kizitonwose** [comment]
Clicks are not detected using `Action.MOVE` event. I disabled scroll in example 5 in the sample app and clicks are still received. See the screen recording below: 

<img src="https://user-images.githubusercontent.com/15170090/62194428-c045ff80-b371-11e9-8e70-2dc2bf8421ed.gif" alt="Example 5" width="270">

**Comment by federicomiralles** [comment]
@kizitonwose I tried the same on my end by adding:
```exFiveCalendar.addOnItemTouchListener(object : RecyclerView.SimpleOnItemTouchListener() {
            override fun onInterceptTouchEvent(rv: RecyclerView, e: MotionEvent): Boolean {
                return if (e.action == MotionEvent.ACTION_MOVE) {
                    true
                } else {
                    super.onInterceptTouchEvent(rv, e)
                }
            }
        })
```
to the example you mentioned but I got the same result... I can't tap on a day... 
I tried using a Motorola One with Android Pie.

**Comment by kizitonwose** [comment]
That's very strange. Can you test with a different phone? 
Also, do you have any other click listeners attached on any part of the calendar? Example, month header or footer?

Sadly, this is a problem with `RecyclerView` itself, not this library so I may not be able to help much.

**Comment by federicomiralles** [comment]
@kizitonwose unfortunately it also happens on Samsung devices but it's strange because it does works on other like Xiaomi. :/

**Comment by kizitonwose** [comment]
Currently waiting to get a Samsung device to test with.

**Comment by kizitonwose** [comment]
I finally got my hands on a Samsung device and can confirm that the suggested workaround does not work on the device. 

However, after further testing, there's another workaround which works on Samsung devices as well as other devices which I tested on:

```kotlin
calendarView.addOnItemTouchListener(object : RecyclerView.SimpleOnItemTouchListener() {
    override fun onInterceptTouchEvent(rv: RecyclerView, e: MotionEvent): Boolean {
        return rv.scrollState == RecyclerView.SCROLL_STATE_DRAGGING
    }
})
```
---

**For reference purposes, here are other things I tried and how they turned out.**

```kotlin
calendarView.setOnTouchListener { _, event ->
    event.action == MotionEvent.ACTION_MOVE
}
```

The above workaround allows an initial scroll action on the calendar before taking effect which is rather weird.

Another option is to extend the `CalenderView` class and override the `dispatchTouchEvent` method.
```kotlin
override fun dispatchTouchEvent(ev: MotionEvent): Boolean {
    return if (ev.action == MotionEvent.ACTION_MOVE) true else super.dispatchTouchEvent(ev)
}
```

This gives the desired result but after a swipe action, while the calendar does not move, a touch event is registered at the position where the swipe gesture started, hence the date cell at that position receives a click event.

**Comment by kizitonwose** [comment]
Closing this issue as there's now a solution. If you need any further assistance, let me know.
